incubator/hnc/internal/reconcilers/anchor.go

```diff
@@ -335,6 +346,16 @@ func (r *AnchorReconciler) writeInstance(ctx context.Context, log logr.Logger, i
 	return nil
 }
 
+// deleteInstance deletes the anchor instance. Note: Make sure there's no
+// finalizers on the instance before calling this function.
+func (r *AnchorReconciler) deleteInstance(ctx context.Context, log logr.Logger, inst *api.SubnamespaceAnchor) error {
+	if err := r.Delete(ctx, inst); err != nil {
+		log.Error(err, "while deleting on apiserver")
+		return err
```

Comment:
We generally don't both log and return errors, right? Usually we do one or the other? An error that's returned to controller-runtime gets logged automatically, so we don't need to log it here.

Maybe: `return fmt.Errorf("while deleting on apiserver: %w", err)`?